executor/instruction/has_executor.rs

```diff
@@ -166,8 +162,15 @@ impl HasExecutor {
         let snapshot = &**context.snapshot();
         let thing_manager = context.thing_manager();
 
-        match self.iterate_mode {
-            BinaryIterateMode::Unbound => {
+        let attribute = self.has.attribute().as_variable().unwrap().as_position();
+        let owner = self.has.owner().as_variable().unwrap().as_position();
+
+        let attribute =
+            attribute.and_then(|pos| (row.len() > pos.as_usize()).then(|| row.get(pos))).filter(|v| !v.is_empty());
+        let owner = owner.and_then(|pos| (row.len() > pos.as_usize()).then(|| row.get(pos))).filter(|v| !v.is_empty());
+
+        match (owner, attribute) {
+            (None, None) => {
```

Comment:
When the planner was planning this - A `has` instruction in a disjunction branch with it's owner `$owner` optionally bound - what would it have considered as the binding state of `$owner`?  Does it consider it unbound?
Else, Wouldn't it pick a fully unbound though there may have been a "slightly more expensive" but always bound plan starting from a different side of the pattern?

Reply:
Yes, it considers it to be unbound.